src/kernels/kernel.cpp

```diff
@@ -0,0 +1,24 @@
+#include "awkward/kernels/kernel.h"
+#include "awkward/cpu-kernels/operations.h"
+#include "awkward/cpu-kernels/util.h"
+#include "awkward/cpu-kernels/getitem.h"
+#include "awkward/cpu-kernels/identities.h"
+#include "awkward/cpu-kernels/reducers.h"
+#include "awkward/gpu-kernels/identities.h"
+
+
+ERROR KernelCore::new_identities64(uint8_t memory_loc, int64_t *toptr, int64_t length) {
+    if (memory_loc < 0)
+        awkward_new_identities64(toptr, length);
+    else {
+        awkwardgpu_new_identities64(memory_loc, toptr, length);
+    }
+}
+
+ERROR KernelCore::new_identities32(uint8_t memory_loc, int32_t *toptr, int64_t length) {
+    if (memory_loc < 0)
+        awkward_new_identities32(toptr, length);
+    else {
+        awkwardgpu_new_identities32(memory_loc, toptr, length);
+    }
+}
```

Comment:
@reikdas's part would be to generate a lot of these.

What do you both think about renaming the `awkward_*` functions to include "cpu" in the name: `awkwardcpu_*`? Symmetry is good, but the "c" vs "g" could get lost when looking at them casually; at least the asymmetry makes it easy to distinguish them from a distance (without glasses). Separating the "cpu" vs "gpu" from the "awkward" with an underscore could help with that as well: `awkward_cpu_*` vs `awkward_gpu_*` are more easily distinguishable.

The primary purpose of this tag is to avoid symbol conflicts. Since the functions are compiled as `extern "C"`, there's nothing that keeps one library's symbols from clashing with another library's symbols except the name. Beyond the fact that the `Kernel` class (or `kernel` namespace) centralizes dispatch between CPU and GPU, it also allows us to remove this prefix, since it's now isolated.

Reply:
Yeah, `awkward_` with a  `cpu_*` or `gpu_*`, should be fine.